pk.go

```diff
@@ -0,0 +1,48 @@
+package ghostferry
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/siddontang/go-mysql/schema"
+)
+
+type PK struct {
+	Value interface{}
+}
+
+func (pk PK) Compare(other PK) int {
+	switch v1 := pk.Value.(type) {
+	case uint64:
+		v2 := other.Value.(uint64)
+
+		if v1 < v2 {
+			return -1
+		} else if v1 == v2 {
+			return 0
+		} else {
+			return 1
+		}
+
+	default:
+		panic(fmt.Sprintf("pk type %T are current not supported", v1))
+	}
+}
+
+func MinPK(t int) PK {
+	switch t {
+	case schema.TYPE_NUMBER:
```

Comment:
If this function is going to depend on siddontang's `schema` then it makes more sense to me that the input argument to it be the `TableColumn` struct itself